fw-child/resources/js/cdc.js

```diff
@@ -781,217 +783,123 @@
-                  current_opacity +
-                  '; border-bottom: 10px solid ' +
-                  unitColor +
-                  '; border-left: 8px solid transparent; border-right: 8px solid transparent"></div>' +
-                  '<span class="legendUnit">&gt; ' +
-                  unitValue +
-                  '</span>' +
-                  '</div>',
-              );
-            } else if (i == colormap.length - 1) {
-              // last row
-              labels.push(
-                '<div class="' +
-                  row_class +
-                  '">' +
-                  '<span class="legendLabel min">' +
-                  unitValue +
-                  '</span>' +
-                  '<div class="legendColor" style="opacity: ' +
-                  current_opacity +
-                  '; border-top: 10px solid ' +
-                  unitColor +
-                  '; border-left: 8px solid transparent; border-right: 8px solid transparent"></div>' +
-                  '<span class="legendUnit">&lt; ' +
-                  unitValue +
-                  '</span>' +
-                  '</div>',
-              );
-            } else {
-              labels.push(
-                '<div class="' +
-                  row_class +
-                  '">' +
-                  '<div class="legendColor" style="opacity: ' +
-                  current_opacity +
-                  '; background-color:' +
-                  unitColor +
-                  ';"></div>' +
-                  '<div class="legendUnit">' +
-                  unitValue +
-                  '</div>' +
-                  '</div>',
-              );
-            }
-          } else {
-            if (i == colormap.length - 1) {
-              // last row if no unit
-
-              labels.push(
-                '<div class="' +
-                  row_class +
-                  ' zero">' +
-                  '<span class="legendLabel ">0</span>' +
-                  '<div class="legendColor" style="opacity: ' +
-                  current_opacity +
-                  '; background-color:' +
-                  unitColor +
-                  '"></div>' +
-                  '<span class="legendUnit">0</span>' +
-                  '</div>',
-              );
-            }
-          }
+        if (typeof callback == 'function') {
+          callback();
         }
 
-        labels.push('</div><!-- .legendRows -->');
 
-        return labels;
       },
 
       get_color: function (d) {
         let plugin = !this.item ? this.data('cdc_app') : this,
           options = plugin.options;
 
-        for (let i = options.legend.colormap.length - 1; i > 0; i--) {
-          if (d < options.legend.colormap[i].quantity) {
-            return options.legend.colormap[i].color;
+        let prev_colour = options.legend.colormap.colours[0],
+          prev_value = options.legend.colormap.quantities[0];
+
+        for (let i = 0; i < options.legend.colormap.colours.length; i++) {
```

Comment:
I will rewrite this with a binary search... it will be around 4 times faster. Since get_color is called very often, it will make a difference.